src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -33,14 +33,20 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(Authentication authentication) {
+        ShoppingCart cart = shoppingCartService.getByUser(userService
+                .findByEmail(authentication.getName())
+                .orElseThrow(() -> new RuntimeException(
+                        "Can't complete order for user with email " + authentication.getName())));
```

Comment:
this exception will appears if there is no user with such email, so lets change the exception message